lib/sqlalchemy/util/typing.py

```diff
@@ -235,6 +236,12 @@ def eval_name_only(
     if "." in name:
         return eval_expression(name, module_name, locals_=locals_)
 
+    # Try builtins first, to handle `list`, `set` or `dict`, etc.
+    try:
+        return builtins.__dict__[name]
```

Comment:
**mike bayer** (zzzeek) wrote:

if the module has its own name "set" or "list" we should look that up first.  will move

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4874